spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/GenericsHelpers.java

```diff
@@ -238,16 +254,17 @@ private static void assertEqualsUnsafe(Types.MapType map, Map<?, ?> expected, Ma
 
     for (int i = 0; i < expectedElements.size(); i += 1) {
       Map.Entry<?, ?> expectedPair = expectedElements.get(i);
-      Object actualKey = actualKeys.get(i, convert(keyType));
-      Object actualValue = actualValues.get(i, convert(keyType));
+      Object actualKey = actualKeys.isNullAt(i) ? null : actualKeys.get(i, convert(keyType));
+      Object actualValue = actualValues.isNullAt(i) ? null : actualValues.get(i, convert(valueType));
```

Comment:
This was the difference between the previous assertions that compare two InternalRow instances and the generic assertions. The internal row version was checking for null using `isNullAt` and the generic ones were not. I think this is most of an error in the tests, but in the case where `get` is called it is good to still return null rather than an incorrect map.